docs/solutions/code-quality/shell-and-or-chain-operator-precedence-bugs.md

```
@@ -0,0 +1,125 @@
+```bash
+[ -n "$SESSION_ID" ] && opencode session delete "$SESSION_ID" 2>/dev/null || warn "cleanup failed"
+```
+
+This chain has two independent failure modes:
+
+### Failure mode 1: Warning fires on empty `$SESSION_ID`
+
+Shell operator precedence: `&&` and `||` are left-associative and equal
+precedence. The chain parses as:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)